modules/mono/glue/GodotSharp/GodotSharp/Core/Bridge/ScriptManagerBridge.cs

```diff
@@ -804,11 +804,11 @@ static void AppendTypeName(StringBuilder sb, Type type)
                 using var methods = new Collections.Array();
 
                 Type? top = scriptType;
-                Type native = GodotObject.InternalGetClassNativeBase(top);
+                Type native = GodotObject.InternalGetClassNativeBase(scriptType);
 
-                while (top != null && top != native)
+                if (scriptType != null && scriptType != native)
```

Comment:
`scriptType` should never be null, so I don't think we need to check.

```suggestion
                if (scriptType != native)
```